src/preferences/configobject.cpp

```diff
@@ -237,10 +237,12 @@ bool ConfigObject<ValueType>::save() {
     }
 
     QFile oldConfig(m_filename);
-    if (!oldConfig.remove()) {
-        qWarning().nospace() << "Could not remove old config file: "
-                             << oldConfig.fileName() << ": " << oldConfig.errorString();
-        return false;
+    if (oldConfig.exists()) {
```

Comment:
Combine into a single if statement

Reply:
I'd prefer to keep them separate. QDir::remove is not just checking something, it has side effects. I think it's easier to understand that by separating these.

Reply:
QFile::exists() also needs to access the file system. I don't see a benefit, short-circuit evaluation is an obvious choice here. 

Nevertheless we should a comment that removal of non-existent files will fail, no matter if we use two or a single if statement.